plugins/hls-class-plugin/src/Ide/Plugin/Class/CodeAction.hs

```diff
@@ -0,0 +1,187 @@
+
+                mkCodeAction title cmd
+                    = InR
+                    $ CodeAction
+                        title
+                        (Just CodeActionQuickFix)
+                        (Just (List []))
+                        Nothing
+                        Nothing
+                        Nothing
+                        (Just cmd)
+                        Nothing
+
+        findClassIdentifier docPath range = do
+            (hieAstResult, pmap) <- handleMaybeM "Unable to GetHieAst"
+                . liftIO
+                . runAction "classplugin.findClassIdentifier.GetHieAst" state
+                $ useWithStale GetHieAst docPath
+            case hieAstResult of
+                HAR {hieAst = hf} ->
+                    pure
+                        $ head . head
+                        $ pointCommand hf (fromJust (fromCurrentRange pmap range) ^. J.start & J.character -~ 1)
+                        ( (Map.keys . Map.filter isClassNodeIdentifier . getNodeIds)
+                            <=< nodeChildren
+                        )
+
+        findClassFromIdentifier docPath (Right name) = do
+            (hscEnv -> hscenv, _) <- handleMaybeM "Unable to GhcSessionDeps"
+                . liftIO
+                . runAction "classplugin.findClassFromIdentifier.GhcSessionDeps" state
+                $ useWithStale GhcSessionDeps docPath
+            (tmrTypechecked -> thisMod, _) <- handleMaybeM "Unable to TypeCheck"
+                . liftIO
+                . runAction "classplugin.findClassFromIdentifier.TypeCheck" state
+                $ useWithStale TypeCheck docPath
+            handleMaybeM "Error in TcEnv"
+                . liftIO
+                . fmap snd
+                . initTcWithGbl hscenv thisMod ghostSpan $ do
+                    tcthing <- tcLookup name
+                    case tcthing of
+                        AGlobal (AConLike (RealDataCon con))
+                            | Just cls <- tyConClass_maybe (dataConOrigTyCon con) -> pure cls
+                        _ -> panic "Ide.Plugin.Class.findClassFromIdentifier"
+        findClassFromIdentifier _ (Left _) = throwE "Ide.Plugin.Class.findClassIdentifier"
+
+isClassNodeIdentifier :: IdentifierDetails a -> Bool
+isClassNodeIdentifier ident = (isNothing . identType) ident && Use `Set.member` identInfo ident
+
+isClassMethodWarning :: T.Text -> Bool
+isClassMethodWarning = T.isPrefixOf "• No explicit implementation for"
+
+-- Return (name text, signature text)
+minDefToMethodGroups :: Range -> [InstanceBindTypeSig] -> BooleanFormula Name -> [[(T.Text, T.Text)]]
+minDefToMethodGroups range sigs = go
+    where
+        go (Var mn)   = [[ (T.pack . occNameString . occName $ mn, bindRendered sig)
+                        | sig <- sigs
+                        , inRange range (getSrcSpan $ bindName sig)
+                        , printOutputable mn == T.drop (T.length bindingPrefix) (printOutputable (bindName sig))
+                        ]]
```

Comment:
Why is this changed to require extra conditions?

Reply:
Because we passed all binds from type check here, should filter valid ones.

Reply:
Oh this is only to pick sig... I was confused. Thanks.